packages/react-core/src/components/Touchspin/Touchspin.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Touchspin/touchspin';
+import { css } from '@patternfly/react-styles';
+import MinusIcon from '@patternfly/react-icons/dist/js/icons/minus-icon';
+import PlusIcon from '@patternfly/react-icons/dist/js/icons/plus-icon';
+import { Button, ButtonProps } from '../Button';
+
+export interface TouchspinProps extends React.HTMLProps<HTMLDivElement> {
+  /** Value of the touchspin */
+  value?: number;
+  /** Additional classes added to the Skeleton */
+  className?: string;
+  /** Sets the width of the touchspin to a number of characters */
+  widthChars?: number;
+  /** Indicates the whole touchspin should be disabled */
+  isDisabled?: boolean;
+  /** Callback for the minus button */
+  onMinus?: (name: string, event: React.MouseEvent) => void;
+  /** Callback for the text input changing */
+  onChange?: (event: React.FormEvent<HTMLInputElement>) => void;
+  /** Callback for the plus button */
+  onPlus?: (name: string, event: React.MouseEvent) => void;
+  /** Adds the given touchspin unit to the touchspin */
+  unit?: React.ReactNode;
```

Comment:
@mcarrano is there a use case for a unit to be something other than a string (e.g, maybe an icon)?

Reply:
@tlabaj I can't really think of why that would be the case, but I'm fine with making this flexible.